devicetypes/smartthings/eaton-accesory-dimmer.src/eaton-accessory-dimmer.groovy

```diff
@@ -0,0 +1,286 @@
+	// creating on/off event is omitted, that state will be updated later (when level drops down to 0)
+	def cmds = []
+	getDelayedGetCommandsDimmerRampTime(cmds)
+	return response(cmds)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd) {
+	createEvent(name: "level", value: cmd.value, unit: "%")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelStartLevelChange cmd) {
+	//Do nothing. Device will send SwitchMultilevelSet when level is set.
+	return null
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	handleDimmerValue(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	handleDimmerValue(cmd.value)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	def delayedOff = settings.delayedOffTime ? settings.delayedOffTime : 10
+	def dimmerRamp = settings.dimmerRampTime ? settings.dimmerRampTime : 5
+	//settings are read-only, so send event if settings are out of sync
+	if(cmd.parameterNumber == 1 && cmd.configurationValue[0] != delayedOff) {
+		createEvent([descriptionText: "$device.displayName delayed off time settings are out of sync. Please save device preferences.",
+				isStateChange: true])
+	} else if (cmd.parameterNumber == 5 && cmd.configurationValue[0] != getPowerUpStateValue()) {
+		createEvent([descriptionText: "$device.displayName power up state settings are out of sync. Please save device preferences.",
+				isStateChange: true])
+	} else if (cmd.parameterNumber == 7 && cmd.configurationValue[0] != dimmerRamp){
+		createEvent([descriptionText: "$device.displayName dimmer ramp time settings are out of sync. Please save device preferences.",
+				isStateChange: true])
+	}else {
+		return null
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	createEvent(descriptionText: "$device.displayName: $cmd", isStateChange: true)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+	log.debug "ManufacturerSpecificReport: ${cmd}"
+	if (cmd.manufacturerName) {
+		updateDataValue("manufacturer", cmd.manufacturerName)
+	}
+	if (cmd.productTypeId) {
+		updateDataValue("productTypeId", cmd.productTypeId.toString())
+	}
+	if (cmd.productId != null) {
+		updateDataValue("productId", cmd.productId.toString())
+	}
+	return null
+}
+
+private handleDimmerValue(newValue){
+	def result = []
+	result << createEvent(name: "switch", value: newValue > 0 ? "on" : "off")
+	result << createEvent(name: "level", value: newValue)
+	return result
+}
+
+private handleOnOff(newValue){
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(zwave.basicV1.basicSet(value: newValue).format())
+	cmds << new physicalgraph.device.HubAction(zwave.basicV1.basicGet().format())
+	sendDelayedCommandsAccordingToDimmerRampTime(cmds)
+}
+
+private sendDelayedCommandsAccordingToDimmerRampTime(cmds){
```

Comment:
As @ALamchaS pointed out to me, this method can get very chatty very quickly. 10 commands in 10s seems like a lot. I prefer his method of just sending one get at the end of the ramp time.